### PR TITLE
[4.2] Contact form banned emails and content

### PR DESCRIPTION
**Pull request by brianteeman**
Adds a trim so that if the entries are in the form example.com; example2.com then example2.com will still be blocked

Improved the description  It is not a list of email addresses but of email addresses or parts of it (e.g. domains).

Pull Request for Issue #39900 .

### Actual result BEFORE applying this Pull Request
the second domain is not banned because of the leading space

### Expected result AFTER applying this Pull Request
both domains are banned

### Updated
the same change has been made to banned words and banned subjects

### Link to documentations
Please select:
- [ ] Documentation link for docs.joomla.org: <link>
- [x] No documentation changes for docs.joomla.org needed

- [ ] Pull Request link for manual.joomla.org: <link>
- [x] No documentation changes for manual.joomla.org needed

### CODE REVIEW

File: components/com_contact/src/Rule/ContactEmailMessageRule.php [hunk 1]
```diff
@@ -47,7 +47,7 @@ public function test(\SimpleXMLElement $element, $value, $group = null, Registry
 
         if ($banned) {
             foreach (explode(';', $banned) as $item) {
-                if ($item != '' && StringHelper::stristr($value, $item) !== false) {
+                if (ltrim($item) != '' && StringHelper::stristr($value, ltrim($item)) !== false) {
```
Comment (Fedik) [inline components/com_contact/src/Rule/ContactEmailMessageRule.php]: Please make it like this:
```php
...
$item = trim($item);
if ($item != '' && StringHelper::stristr($value, $item) !== false) {
...
```

File: administrator/language/en-GB/com_contact.ini [hunk 2]
```diff
@@ -26,7 +26,7 @@ COM_CONTACT_FIELD_ARTICLES_DISPLAY_NUM_LABEL="# Articles to List"
 COM_CONTACT_FIELD_ARTICLES_SHOW_LABEL="User Articles"
 COM_CONTACT_FIELD_CAPTCHA_LABEL="Allow Captcha on Contact"
 COM_CONTACT_FIELD_CATEGORY_LABEL="Select a Category"
-COM_CONTACT_FIELD_CONFIG_BANNED_EMAIL_DESC="Separate multiple email addresses with a semicolon."
+COM_CONTACT_FIELD_CONFIG_BANNED_EMAIL_DESC="Separate multiple email addresses with a semicolon. Emails from a complete domain can be banned by using example.com."
```
Comment (Fedik) [inline administrator/language/en-GB/com_contact.ini]: Complete domain is `@example.com` and `example.com` will be a "partial", for email here.

File: components/com_contact/src/Rule/ContactEmailMessageRule.php [hunk 3]
```diff
@@ -47,6 +47,7 @@ public function test(\SimpleXMLElement $element, $value, $group = null, Registry
 
         if ($banned) {
             foreach (explode(';', $banned) as $item) {
+                $item = trim($item);
```
Comment (sandewt) [inline components/com_contact/src/Rule/ContactEmailMessageRule.php:50]: > Adds an ltrim so that if the entries

`ltrim` or `trim` ?

Comment (brianteeman) [reply components/com_contact/src/Rule/ContactEmailMessageRule.php:50]: I did it as ltrim originally but then fedik suggested to use trim (remove spaces at both start and end) and I ddint update the description (doing that now)

Comment (sandewt) [reply components/com_contact/src/Rule/ContactEmailMessageRule.php:50]: OK

**Comment by Fedik** [comment]
I have tested this item :white_check_mark: successfully on 8737f9afc3e1ccb86354fd98284b43187af9a391<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/39907">issues.joomla.org/tracker/joomla-cms/39907</a>.</sub>

**Comment by sandewt** [comment]
Do the following files also need to be changed?

https://github.com/joomla/joomla-cms/blob/56c6ad360c3e70c060bb99026c8f92315ca41d3c/components/com_contact/src/Rule/ContactEmailRule.php#L52-L58

https://github.com/joomla/joomla-cms/blob/56c6ad360c3e70c060bb99026c8f92315ca41d3c/components/com_contact/src/Rule/ContactEmailSubjectRule.php#L48-L54

**Comment by brianteeman** [comment]
you can change them if you want in their own pr. this pr is about the email address

**Comment by sandewt** [comment]
> this pr is about the email address

You changed the file of the banned words

https://github.com/joomla/joomla-cms/blob/8737f9afc3e1ccb86354fd98284b43187af9a391/components/com_contact/src/Rule/ContactEmailMessageRule.php#L31

**Comment by Fedik** [comment]
@sandewt is right, @brianteeman can you please edit also ContactEmailRule.php

**Comment by brianteeman** [comment]
oh - shoot - changed the wrong file ;)

**Comment by sandewt** [comment]
I have tested this item :white_check_mark: successfully on 856ba4dbbf8299f6b8ae0d80784fa05614fbbc09<br /><br />Tested:

1) Banned Email Address
2) Banned Subject
3) Banned Text<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/39907">issues.joomla.org/tracker/joomla-cms/39907</a>.</sub>

**Comment by sandewt** [comment]
@brianteeman I think it would be useful to expand the test description, see my previous comment above

**Comment by viocassel** [comment]
I have tested this item :white_check_mark: successfully on 856ba4dbbf8299f6b8ae0d80784fa05614fbbc09<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/39907">issues.joomla.org/tracker/joomla-cms/39907</a>.</sub>

**Comment by alikon** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/39907">issues.joomla.org/tracker/joomla-cms/39907</a>.</sub>

**Comment by roland-d** [comment]
Thank you.

**Comment by brianteeman** [comment]
thanks

**Pull request merged**